bridge_adaptivity/module/engines/engine_vpal.py

```diff
@@ -0,0 +1,138 @@
+import logging
+import requests
+import urlparse
+
+from module.engines.interface import EngineInterface
+
+log = logging.getLogger(__name__)
+
+
+ACTIVITY_PARAMETERS = (
+    'collection',
+    'name',
+    'tags',
+    'type',
+    'difficulty',
+)
+
+SEQUENCE_ITEM_PARAMS = (
+    'learner',
+    'activity',
+    'score',
+)
+
+TYPES = {
+    'A': 'pre-assessment',
+    'Z': 'post-assessment'
+}
+
+
+class EngineVPAL(EngineInterface):
+    """
+    VPAL adaptive engine.
+
+    To use VPAL engine with the Bridge for Adaptivity configure following parameters in the secure settings::
+
+        ENGINE_MODULE = 'module.engines.engine_vpal'
+
+        # ENGINE_DRIVER is a string with the name of driver class in the engine module
+        ENGINE_DRIVER = 'EngineVPAL'
+
+        # ENGINE_SETTINGS is a dict, with the initial params for driver initialization
+        ENGINE_SETTINGS = {
+            'BASE_URL': 'http://test.engine.vpal.io/engine/api',
+        }
+    """
+    def __init__(self, **kwargs):
+        self.host = kwargs.get('HOST')
+        self.api_url = 'engine/api/'
+        self.base_url = urlparse.urljoin(self.host, self.api_url)
+        self.activity_url = urlparse.urljoin(self.base_url, "activity")
+
+    @staticmethod
+    def check_engine_response(engine_status, action=None, activity_name=None, status=200):
+        if engine_status == status:
+            log.debug("[VPAL Engine] activity {} is {} to the VPAL Engine.".format(activity_name, action))
+            return True
+        else:
+            log.error("[VPAL Engine] activity {} is not {} to the VPAL Engine.".format(activity_name, action))
+            return False
+
+    @staticmethod
+    def fulfil_payload(payload={}, instance_to_parse=None):
+        from module.models import Activity, SequenceItem
+        if isinstance(instance_to_parse, Activity):
+            params = ACTIVITY_PARAMETERS
+        elif isinstance(instance_to_parse, SequenceItem):
+            params = SEQUENCE_ITEM_PARAMS
+        else:
+            raise ValueError("Payload for the instance {} cannot be prepared".format(instance_to_parse))
+        for param in params:
+            if param == 'type':
+                atype = TYPES.get(getattr(instance_to_parse, 'atype'), 'generic')
+                payload[param] = atype
```

Comment:
does this give activity field "atype" to the engine as "type"?

Reply:
Yes, it is.